detect-stale-job/action.yml

```diff
@@ -0,0 +1,82 @@
+name: Detect stale job
+description: Detect if a newer workflow run already has run the current job - making the current run stale. The job needs 'write' permissions to 'actions' in order to cancel a run.
+
+runs:
+  using: composite
+  steps:
+    - name: Cancel if stale
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+        CHECK_RUN_ID: ${{ job.check_run_id }}
+      run: |
+        current_sha="$GITHUB_SHA"
+        current_run_id="$GITHUB_RUN_ID"
+
+        # Get current job details
+        current_job_details="$(gh run view --repo "$GITHUB_REPOSITORY" "$GITHUB_RUN_ID" --json jobs | jq --argjson check_run_id "$CHECK_RUN_ID" '.jobs[] | select(.databaseId == $check_run_id)')"
+        current_job_name="$(echo "$current_job_details" | jq --exit-status --raw-output '.name')"
+        current_job_started_at="$(echo "$current_job_details" | jq --exit-status --raw-output '.startedAt')"
+
+        echo "Current SHA: $current_sha"
+        echo "Current run ID: $current_run_id"
+        echo "Current job: $current_job_name"
+        echo "Current job started at: $current_job_started_at"
+        echo ""
+
+        # NOTE: Run numbers are not guaranteed to be in order of commits, but it's good enough for most cases.
+        # For more accuracy, you could use: git merge-base --is-ancestor "$GITHUB_SHA" "$SHA_OF_POTENTIAL_DESCENDANT"
+        # but that would require fetching git history which may be slow.
+
+        # Get list of workflow runs for current workflow on current branch with a run number greater than the current run
+        runs="$(gh run list \
+          --repo "$GITHUB_REPOSITORY" \
+          --workflow "$GITHUB_WORKFLOW" \
+          --branch "$GITHUB_REF_NAME" \
+          --json "databaseId,number" \
+          --limit 20 \
+          | jq --raw-output --argjson current_run_number "$GITHUB_RUN_NUMBER" '.[] | select(.number > $current_run_number) | .databaseId')"
+
+        if [ "$runs" == "" ]; then
+          echo "No newer runs found - current run is up to date"
+          exit 0
+        fi
+
+        echo "Checking if any newer runs have progressed further than the current run:"
+        echo "$runs"
+        echo ""
+
+        echo "$runs" | while read -r run_id; do
+          if [ "$run_id" == "" ]; then
+            continue
+          fi
+
+          # NOTE: Checking if the newer run has reached or passed the job, making the current run out-of-order.
+          # We look for a newer job that is in progress, or completed and has run at least one step (i.e. not skipped).
+          newer_run_jobs="$(gh run view --repo "$GITHUB_REPOSITORY" "$run_id" --json jobs)"
```

Comment:
Vurder å skriv ned her eller et annet steg hvorfor vi gjør dette, og ikke kun gjør sjekken på linje 40.

Dvs man er i denne situasjonen:
- Run 1 venter på manuell godkjenning til deploy til prod
- Run 2 står som pending på deploy til prod, med concurrent lås fra GitHub
- Bruker trykke approve på manuell godkjening
- Uten denne sjekken, så vil koden finne en ny run (run 2), og avbryte. For brukeren så vil man oppleve at man godkjenner en run, også blir den bare avbrutt fordi det finnes en nyere run, som er litt dust, ingen har lyst å godkjenne noe som bare blir avbrutt.

Reply:
Hvis vi skriver det et annet sted, hadde det vært en fint med en lenke i kommentar eller README eller noe i denne kodebasen, så det er lett å finne.

Reply:
Veldig godt poeng! La til noen kommentarer i koden nå. Ser det greit ut?